spec/app/ics-027-interchain-accounts/README.md

```diff
@@ -501,34 +498,29 @@ function onChanCloseConfirm(
 
 ```typescript
 function OnRecvPacket(packet Packet) {
-    	ack := NewResultAcknowledgement([]byte{byte(1)})
+    	ack = NewResultAcknowledgement([]byte{byte(1)})
 
 	// only attempt the application logic if the packet data
 	// was successfully decoded
-	if ack.Success() {
-		switch data.Type {
-	          case types.EXECUTE_TX:
-		        msgs, err := types.DeserializeTx(data.Data)
-		        if err != nil {
-			      return err
-		        }
-
-                        // ExecuteTx calls the AuthenticateTx function defined above 
-		        if err = ExecuteTx(ctx, packet.SourcePort, packet.DestinationPort, packet.DestinationChannel, msgs); err != nil {
-			      return err
-		        }
-
-		        return nil
-	          default:
-		        return ErrUnknownDataType
-	        }
-	        if err != nil {
-                    ack = NewErrorAcknowledgement(err.Error())
-                }
-	}
-
-	// NOTE: acknowledgment will be written synchronously during IBC handler execution.
-	return ack
+  switch data.Type {
+  case types.EXECUTE_TX:
+  msgs, err := types.DeserializeTx(data.Data)
+  if err != nil {
+    return NewErrorAcknowledgement(err)
```

Comment:
should the spec discuss the implication of returning the error in the acknowledgement? That is, including the error string in the acknowledgement makes error strings state machine breaking

Reply:
Maybe this should be optional, as we will not include the error in the ibc-go implementation

Reply:
> should the spec discuss the implication of returning the error in the acknowledgement? That is, including the error string in the acknowledgement makes error strings state machine breaking

Very good point.
That has proved to be the case for CosmWasm with submessages and we did hit a case where one node included the stack trace and fell out of consensus.

I like to include error message, but this is a very good thing to think through (and check code paths)